sonic-utilities-tests/feature_test.py

```diff
@@ -0,0 +1,165 @@
+import sys
+import os
+import mock
+from click.testing import CliRunner
+from swsssdk import ConfigDBConnector
+
+test_path = os.path.dirname(os.path.abspath(__file__))
+modules_path = os.path.dirname(test_path)
+sys.path.insert(0, modules_path)
+
+import show.main as show
+import config.main as config
+
+config.asic_type = mock.MagicMock(return_value = "broadcom")
+config._get_device_type = mock.MagicMock(return_value = "ToRRouter")
+
+config_db = ConfigDBConnector()
+config_db.connect()
+
+show.config_db = config_db
+config.config_db = config_db
+
+show_feature_status_output="""\
+Feature     State     AutoRestart
+----------  --------  -------------
+bgp         enabled   enabled
+database    enabled   disabled
+dhcp_relay  enabled   enabled
+lldp        enabled   enabled
+nat         enabled   enabled
+pmon        enabled   enabled
+radv        enabled   enabled
+restapi     disabled  enabled
+sflow       disabled  enabled
+snmp        enabled   enabled
+swss        enabled   enabled
+syncd       enabled   enabled
+teamd       enabled   enabled
+telemetry   enabled   enabled
+"""
+
+show_feature_bgp_status_output="""\
+Feature    State    AutoRestart
+---------  -------  -------------
+bgp        enabled  enabled
+"""
+
+show_feature_bgp_disabled_status_output="""\
+Feature    State     AutoRestart
+---------  --------  -------------
+bgp        disabled  enabled
+"""
+
+show_feature_autorestart_output="""\
+Feature     AutoRestart
+----------  -------------
+bgp         enabled
+database    disabled
+dhcp_relay  enabled
+lldp        enabled
+nat         enabled
+pmon        enabled
+radv        enabled
+restapi     enabled
+sflow       enabled
+snmp        enabled
+swss        enabled
+syncd       enabled
+teamd       enabled
+telemetry   enabled
+"""
+
+show_feature_bgp_autorestart_output="""\
+Feature    AutoRestart
+---------  -------------
+bgp        enabled
+"""
+
+
+show_feature_bgp_disabled_autorestart_output="""\
+Feature    AutoRestart
+---------  -------------
+bgp        disabled
+"""
+
+class TestFeature(object):
+    @classmethod
+    def setup_class(cls):
+        print("SETUP")
+
+    def test_show_feature_status(self):
+        runner = CliRunner()
```

Comment:
Suggest moving this instantialization to a setUp method and only performing it once:

```python
    def setUp(self):
        self.runner = CliRunner()
```

Reply:
setUp method is defined as class method. we cannot put it there.

Reply:
setUp is not the pytest way to run it, setup is old python unittest way, py.test way is to use fixture. actually, all setUp method should be clean up in previous test and we need to add fixture to do it. I suggest to do this in another pr. won't plan to fix this in this pr.